### ISSUE TITLE
Create NPC

### ISSUE THREAD

**Issue by carolinaaraujo00** [carolinaaraujo00/js_prototype#1]
- [x] Create NPC with NavmeshAgent to allow for movement 
- [x] Create an interaction module that allows for interaction when the player is near a NPC that has this module

As per Soni et al., A Framework of Touchscreen Interaction Design Recommendations for Children, 2019
- [x] Interaction should have a big enough area to allow for clicking and pointing errors
- [x] Visual representation of them being clickable 
- [ ] Provide means to reverse actions

**Comment by carolinaaraujo00** [comment]
Animate in relation to the parent node: https://forum.godotengine.org/t/problems-animating-object-position/18282/2
